test/Elastic.Apm.Extensions.Hosting.Tests/HostBuilderExtensionTests.cs

```diff
@@ -24,9 +27,49 @@ public async Task TwoHostBuildersNoException()
 			await Task.WhenAll(hostBuilder1.StopAsync(), hostBuilder2.StopAsync());
 		}
 
+		/// <summary>
+		/// Makes sure that <see cref="Agent.IsConfigured" /> is <code>true</code> after the agent is enabled through
+		/// <see cref="HostBuilderExtensions.UseElasticApm" />.
+		/// </summary>
+		[Fact]
+		public void IsAgentInitializedAfterUseElasticApm()
+		{
+			var _ = CreateHostBuilder().Build();
+			Agent.IsConfigured.Should().BeTrue();
+		}
+
+		/// <summary>
+		/// Makes sure that agent enables the <see cref="IDiagnosticsSubscriber" /> passed into
+		/// <see cref="HostBuilderExtensions.UseElasticApm" />.
+		/// </summary>
+		[Fact]
+		public void DiagnosticSubscriberWithUseElasticApm()
+		{
+			var fakeSubscriber = new FakeSubscriber();
+			fakeSubscriber.IsSubscribed.Should().BeFalse();
+
+			Host.CreateDefaultBuilder()
+				.ConfigureServices((context, services) => { services.AddHostedService<HostedService>(); })
+				.UseElasticApm(fakeSubscriber)
+				.Build();
+
+			fakeSubscriber.IsSubscribed.Should().BeTrue();
+		}
```

Comment:
Should there also be a test to assert that subscribers are not subscribed when the agent is not enabled?

Reply:
Agreed - I added a test.